contracts/Vault.sol

```diff
@@ -161,7 +161,7 @@ contract Vault is
             "Vault: invalid vault"
         );
         require(
-            address(strategy) == address(0) || strategy.investedAssets() == 0,
+            address(strategy) == address(0) || strategy.hasAssets() == false,
```

Comment:
What would be the issue if we just dropped the condition instead?

Reply:
The tx should be fail if it is unable to set new strategy.

Reply:
Sure, but what I'm asking is, what if we don't check if the strategy has assets?

Reply:
the funds in previous strategy will be lost.